shuup/front/basket/__init__.py

```diff
@@ -27,15 +27,15 @@ def get_basket_command_dispatcher(request):
     return cached_load("SHUUP_BASKET_COMMAND_DISPATCHER_SPEC")(request=request)
 
 
-def get_basket(request):
+def get_basket(request, basket_name="basket", basket_class=None):
     """
     :type request: django.http.request.HttpRequest
     :rtype: shuup.front.basket.objects.BaseBasket
     """
-    if not hasattr(request, "basket"):
+    if basket_name == "basket" and hasattr(request, "basket"):
```

Comment:
So if the basket has name and is passed as a parameter new basket is created every time?